trino/trino.go

```diff
@@ -1253,13 +1264,263 @@ type queryResponse struct {
 	PartialCancelURI string        `json:"partialCancelUri"`
 	NextURI          string        `json:"nextUri"`
 	Columns          []queryColumn `json:"columns"`
-	Data             []queryData   `json:"data"`
+	Data             interface{}   `json:"data"`
 	Stats            stmtStats     `json:"stats"`
 	Error            ErrTrino      `json:"error"`
 	UpdateType       string        `json:"updateType"`
 	UpdateCount      int64         `json:"updateCount"`
 }
 
+type spoolingProtocol struct {
+	httpClient http.Client
+	ctx        context.Context
+	encoding   string
+	segments   []interface{}
+}
+
+type spoolingMetadata struct {
+	RowOffset        int64
+	RowsCount        int64
+	SegmentSize      int64
+	UncompressedSize int64
+}
+
+func parseSpoolingMetadata(metadata map[string]interface{}) (spoolingMetadata, error) {
+	var mt spoolingMetadata
```

Comment:
Use more meaningful variable names, not just abbreviations.
```suggestion
	var result spoolingMetadata
```